### PR TITLE
Mock functions for i18n

### PR DESCRIPTION
**Pull request by Hyperkid123**
Created a simple `translate` and `sprintf` function, that should act as a proxy for translation. These functions will work only in environment, where `__` and `sprintf` functions are available on window (as in ManageIQ). If they are not available, it will return original string or mask.

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@martinpovolny can you check it out?

**Comment by martinpovolny** [comment]
@mzazrivec : what do you think?

**Comment by himdel** [comment]
This doesn't work.

Gettext needs that `__(...)` to find the strings to extract.

Wrapping that in a helper function just means we will have no strings..

**Comment by himdel** [comment]
(`sprintf` is just fine, but the "mock" `translate` needs to be called `__`, not `translate`)

**Comment by mzazrivec** [comment]
@Hyperkid123 The `translate()` routine needs to be renamed to `__()`. We don't want to be using something special here.

**Comment by Hyperkid123** [comment]
@mzazrivec ok i will rename it
@himdel the helper function seems to be working just fine. I have just tested it and this is the result. This source 
```
import React from 'react';
import { translate, sprintf } from '../global-functions';

export const TranslateDemo = () => (
  <div>
    <h1>Translated string 1: {sprintf(translate('%d%% in use of %d total'), 15, 50)}</h1>
    <h1>Translated string 2: {translate('Submit')}</h1>
    ## next line won't work in react-ui-components dev server
    <h1>Translated string 3: {__('Submit')}</h1>
  </div>
);
```
will render this is manageIQ:

screenshot from 2018-06-07 09-33-32
Is there something i am missing?

File: demo/demo-app.jsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import * as React from 'react';
 import { Form, Field } from 'react-final-form';
 import { Form as PfForm, Col, Row, Button, Grid } from 'patternfly-react';
 import { FinalFormSwitch } from '../src/forms';
+import { translate, sprintf } from '../src/global-functions';
```
Comment (mzazrivec) [inline demo/demo-app.jsx]: This should read `import { __, sprintf } ...`, right?

Comment (Hyperkid123) [reply demo/demo-app.jsx]: @mzazrivec oh this file should not even be in this PR. Thanks

**Comment by martinpovolny** [comment]
> Is there something i am missing?

Your stuff works in run time, but you are missing the part where the strings are collected for translation. This is done by parsing the sources and collecting all the calls to `__(args)` and a several other functions.

**Comment by Hyperkid123** [comment]
@martinpovolny yes @himdel already explained.

**Comment by Hyperkid123** [comment]
@martinpovolny i have renamed the translate function to `__(...)` and tested it in ManageIQ (even non existing keys) and it seems to be working. When this is merged i think i have all i need to start working on the provider forms.

**Comment by Hyperkid123** [comment]
@martinpovolny @karelhala @himdel can we get this merged? Or is there something else you want me to do?

**Pull request merged**